docs/proposals/container-runtime-interface/capabilities.md

```diff
@@ -0,0 +1,134 @@
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Linux Capabilities in Container Runtime Interface
+
+The following proposal summarizes the current state and problems for runtime to implement Linux Capabilities.
+And propose solutions to solve the problems
+
+## Problems
+
+Today's [Linux Capabilities Interface](https://github.com/kubernetes/kubernetes/blob/v1.5.0-alpha.0/pkg/kubelet/api/v1alpha1/runtime/api.proto#L350-L355) in CRI is derived from the Docker runtime implementation, which also maps
+to the [k8s API](https://github.com/kubernetes/kubernetes/blob/v1.5.0-alpha.0/pkg/api/v1/types.go#L1164-L1170).
+
+If a user wants to launch a container with required capabilities, he will go through such steps:
+
+- Get a list of capabilities he would require
+- Look up Docker's docs about the [default capability list](https://docs.docker.com/engine/reference/run/#/runtime-privilege-and-linux-capabilities)
+- Calculate the `Add` and `Drop` capability set based on the his requirement and the default capabilities list.
+- Specify the `Add` and `Drop` set for the containers in a k8s pod spec.
+- Launch the pod/container.
+
+There are several issues here:
+
+- The default capability set heavily depends on Docker's implementation, Kubernetes has no control over that.
+When Docker changes the default capability set across versions (hopefull it should not happen), users of Dockers get surprise,
+users of Kubernetes get surprise, too.
+- Even worse, to implement the same semantics as today's Docker runtime, it will require other runtimes (e.g. rkt) to depend
+on the default capability set as well. And if Docker changes the default capability set, then consistency is not guaranteed.
+- It is arguable that `Add` and `Drop` set are the best semantics for capability set compared to a [whitelist based](https://github.com/appc/spec/blob/af31bda9a474bf3a3e83144f6f0d264cfedc813f/spec/ace.md#oslinuxcapabilities-retain-set) semantics,
+which explicitly sets the bounding capabilities that a container could have, and has less security implications (users don't guess what
+is the final capability set).
+
+## Solutions
+
+As mentioned above, an explicit whitelist type of capability-retain-set could make things very consistent across runtimes and
+version updates.
+But as we can see, it's not very practical for system admins and operators to put a long [default capability list](https://docs.docker.com/engine/reference/run/#/runtime-privilege-and-linux-capabilities) on every
+container he wants to launch.
+Besides, changing the k8s API could take several release cycles.
+
+So in the proposed solution here. We will:
+- Keep the k8s API as is today. Users will not need to change anything.
+- Define a default capability list inside k8s, and document about it.
```

Comment:
This list should ideally be applied by default across most kubernetes products like minikube, vanilla k8s, etc.
That way a k8s app developer can have a seamless experience across Kubernetes instead of having to fine tune their app for all deployments.